pkg/arvo/app/chat-hook.hoon

```diff
@@ -0,0 +1,364 @@
+::  chat-hook:
+::  mirror chat data from foreign to local based on read permissions
+::  allow sending chat messages to foreign paths based on write perms
+::
+/-  *permission-store, *chat-hook
+/+  *chat-json
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%chat-update chat-update]]
+      [%quit ~]
+      [%poke wire dock poke]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  state-zero
+  $:  synced=(map path ship)
+      boned=(map wire (list bone))
+  ==
+::
++$  poke
+  $%  [%chat-action chat-action]
+      [%permission-action permission-action]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    :_  this
+    [ost.bol %peer /permissions [our.bol %permission-store] /updates]~
+  [~ this(+<+ u.old)]
+::
+++  poke-noun
+  |=  a=*
+  ^-  (quip move _this)
+  ~&  synced
+  ~&  boned
+  [~ this]
```

Comment:
If this arm is just for debug printfs, it'd be better to require an explicit `%debug` or some-such payload.

Also, it's worth reiterating that `~&` is only for developer debug printfs. If the user needs to see something, that should be produced as a move (a `%flog` to %dill is simplest -- some notification channel remains desirable).

Reply:
Thanks for noting. Removed the poke-noun arms. The other ~& references stayed for debugging purposes if users run into problems. I should look more into %flog.

Reply:
@loganallenc, the `+poke-noun` arms appear to still be present.

Reply:
Thanks for mentioning. The commit I just pushed removed them.